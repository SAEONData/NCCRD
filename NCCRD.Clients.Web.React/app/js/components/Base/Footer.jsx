'use strict'

import React from 'react'
import { Col, Row, Container, Footer as MDBFooter, Modal, ModalBody, ModalHeader, ModalFooter } from 'mdbreact';

//Images
import nrf_seaon from '../../../images/nrf_saeon.png'

class Footer extends React.Component {

  constructor(props) {
    super(props);

    this.toggleModal = this.toggleModal.bind(this)

    this.state = { showModal: false }
  }

  toggleModal() {
    this.setState({ showModal: !this.state.showModal })
  }

  render() {

    return (
      <>
        <MDBFooter color="light" style={{ color: "black" }}>
          <hr />
          <Container>
            <Row className="align-items-center" style={{ marginBottom: "10px", marginTop: "10px" }}>
              <Col sm="9">
                <p>
                  <b>Created by SAEON using React, MDBootstrap and AntDesign</b>
                  <br />
                  &copy; {(new Date().getFullYear())} Copyright:
                  <a href="http://www.saeon.ac.za" style={{ color: "black" }}>
                    <b> <u>SAEON.ac.za</u> </b>
                  </a>
                </p>
              </Col>
              <Col md="3" className="d-none d-md-block">
                <img onClick={this.toggleModal} src={nrf_seaon} style={{ width: "80%", marginTop: "-8px", cursor: "pointer" }} align="right" />
              </Col>
            </Row>
          </Container>
        </MDBFooter>

        <Container>
          <Modal isOpen={this.state.showModal} toggle={this.toggleModal} size="fluid" style={{ width: "80%" }} >
            <ModalHeader toggle={this.toggleModal}>What SAEON offers:</ModalHeader>
            <ModalBody>
              <iframe
                style={{
                  width: "100%",
                  height: "500px",
                  margin: "0px",
                  border: "1px solid gainsboro"
                }}
                src={"http://www.example.com"}
              />
            </ModalBody>
          </Modal>
        </Container>
      </>
    )
  }
}

export default Footer