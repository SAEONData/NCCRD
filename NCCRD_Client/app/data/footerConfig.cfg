export const footerConfig = [
  { text: "Terms & Conditions", link: "http://www.example.com" },
  { text: "Contact", link: "http://www.example.com" }
]