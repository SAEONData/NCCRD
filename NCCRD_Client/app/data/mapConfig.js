export const MapConfig = {
  service: "http://app01.saeon.ac.za/nccrdapi/odata/projects/extensions.geojson",
  toolTipField: "name",
  styleField: "status",
  styles: [
      {
          value: null,
          icon: "http://app01.saeon.ac.za/nccrd/yellow.png",
          anchorX: 8,
          anchorY: 8
      },
      {
          value: "ok",
          icon: "http://app01.saeon.ac.za/nccrd/green.png",
          anchorX: 8,
          anchorY: 8
      },
      {
          value: "bad",
          icon: "http://app01.saeon.ac.za/nccrd/red.png",
          anchorX: 8,
          anchorY: 8
      }
  ]
}