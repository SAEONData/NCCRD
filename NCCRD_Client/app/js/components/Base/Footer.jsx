import React from 'react'
import { Col, Row, Container, Footer as MDBFooter, Modal, ModalBody, ModalHeader, ModalFooter } from 'mdbreact';

import { footerConfig } from '../../../data/footerConfig.cfg'

//Images
import nrf_seaon from '../../../images/nrf_saeon.png'

class Footer extends React.Component {

  constructor(props) {
    super(props);

    this.toggleModal = this.toggleModal.bind(this)
    this.renderLinks = this.renderLinks.bind(this)
    this.handleResize = this.handleResize.bind(this)

    this.state = {
      showModal: false,
      modalHeader: "",
      modalSrc: "",
      isNarrowDisplay: false
    }
  }

  componentDidMount(){
    window.addEventListener("resize", this.handleResize);
  }

  componentWillUnmount() {
    window.removeEventListener("resize", this.handleResize)
  }

  handleResize(){
    this.setState({
      isNarrowDisplay: (window.innerWidth < 900)
    })
  }

  toggleModal(state, header = "", src = "") {
    this.setState({
      showModal: state,
      modalHeader: header,
      modalSrc: src
    })
  }

  renderLinks() {

    let footerLinks = []

    if (footerConfig && footerConfig.length > 0) {
      footerConfig.forEach(x => {
        footerLinks.push(
          <div
            key={x.text}
            style={{
              marginLeft: "10px",
              marginRight: "10px",
              color: "#4285F4",
              cursor: "pointer",
              display: "inline-block"
            }}
            onClick={() => this.toggleModal(true, x.text, x.link)}
          >
            <u>{x.text}</u>
          </div>
        )
      })
    }

    return footerLinks
  }

  render() {

    let { showModal, modalHeader, modalSrc, isNarrowDisplay } = this.state

    return (
      <div style={{ padding: "5px 15px 5px 15px", borderTop: "1px solid #E8E8E8" }}>
        <MDBFooter color="light" style={{ color: "black" }}>
          <Row className="align-items-center" style={{ marginBottom: "15px", marginTop: "-10px" }}>

            <Col md="8">
              {this.renderLinks()}
            </Col>

            <Col md="4">
              <br className="d-block d-md-none" />

              <div style={{ float: "right" }}>
                <span style={{ marginLeft: "10px", fontSize: "14px", color: "dimgrey" }}>
                  Copyright &copy; {(new Date().getFullYear())}
                </span>

                {
                  (isNarrowDisplay === true) &&
                  <div style={{ height: "15px" }} />
                }

                <img
                  // onClick={() => this.toggleModal(true, "What SAEON Offers", "http://www.example.com")}
                  src={nrf_seaon}
                  style={{
                    height: "40px",
                    marginTop: "-8px",
                    marginBottom: "-10px",
                    marginLeft: "20px",
                    // cursor: "pointer",
                    border: "0px solid grey"
                  }}
                  align="right"
                />
              </div>
            </Col>

          </Row>
        </MDBFooter>

        <Modal isOpen={showModal} toggle={() => this.toggleModal(false)} size="fluid" style={{ width: "95%" }} >
          <ModalHeader toggle={() => this.toggleModal(false)}>{modalHeader}</ModalHeader>
          <ModalBody>
            <iframe
              style={{
                width: "100%",
                height: "500px",
                margin: "0px",
                border: "1px solid gainsboro"
              }}
              src={modalSrc}
            />
          </ModalBody>
        </Modal>
      </div>
    )
  }
}

export default Footer