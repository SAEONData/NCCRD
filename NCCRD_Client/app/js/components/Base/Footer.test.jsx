import React from "react"
import { mount, shallow } from "enzyme"
import { Col, Row, Modal, ModalBody } from 'mdbreact';
import Footer from "./Footer.jsx"
import { footerContent } from '../../../data/footerConfig'
import { link } from "fs";

describe("Footer", () => {
  let props
  let mountedFooter
  const footer = () => {
    if (!mountedFooter) {
      mountedFooter = mount(
        <Footer />
      )
    }
    return mountedFooter
  }

  beforeEach(() => {
    props = {}
    mountedFooter = undefined
  })

  // All tests will go here
  it('should render correctly with no props', () => {
    expect(footer()).toMatchSnapshot()
  })

  it("config data exists", () => {
    expect(footerContent).toBeDefined()
  })

  it("has 2 rows", () => {
    expect(footer().find(Row).length).toBe(2)
  })

  describe("top/first row", () => {

    //Count the number of sections and text links
    let sectionCount = 0
    let linkCount = 0
    if (footerContent) {
      footerContent.sections.forEach(section => {
        sectionCount += 1
        if (section.links) {
          section.links.forEach(link => {
            if (link.text) {
              linkCount += 1
            }
          })
        }
      })
    }

    it(`has ${sectionCount} columns`, () => {
      expect(footer().find(Row).first().find(Col).length).toBe(sectionCount)
    })

    it(`has ${linkCount} links`, () => {
      expect(footer().find(".custom_link").length).toBe(linkCount)
    })
  })

  describe("bottom/second row", () => {
    it("has exactly one column", () => {
      expect(footer().find(Row).last().find(Col).length).toBe(1)
    })
  })

  it("has exactly one modal defined", () => {
    expect(footer().find(Modal).length).toBe(1)
  })

  //can't seem to test modal content with mount, but shallow works
  describe("modal", () => {
    const component = shallow(<Footer />)
    it("has exactly one iframe", () => {
      expect(component.find("iframe").length).toBe(1)
    })
  })

  it("has at least one image", () => {
    expect(footer().find("img").length).toBeGreaterThan(0)
  })

})