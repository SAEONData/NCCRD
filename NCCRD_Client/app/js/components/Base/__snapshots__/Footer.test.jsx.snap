// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly with no props 1`] = `
<Footer>
  <div
    style={
      Object {
        "backgroundColor": "white",
        "borderLeft": "1px solid gainsboro",
        "borderRight": "1px solid gainsboro",
        "borderTop": "1px solid gainsboro",
        "color": "black",
        "padding": "5px 45px 5px 45px",
      }
    }
  >
    <br />
    <t
      tag="div"
    >
      <div
        className="row"
      >
        <t
          key="section_Technology"
          lg={null}
          md={null}
          sm={null}
          tag="div"
          xl={null}
          xs={null}
        >
          <div
            className="col"
          >
            <h4
              style={
                Object {
                  "marginBottom": "15px",
                }
              }
            >
              <b>
                Technology
              </b>
            </h4>
            <div
              className="custom_link"
              key="link_0"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "default",
                  "fontWeight": "regular",
                }
              }
            >
              DEA NCCRD uses open source, government funded facilities provided by SAEON. The DST and NRF funds the SAEON Open Data Platform (ODP) and associated portals. Developed by SAEON on behalf of DST, DEA, DRDLR, and other stakeholders.
            </div>
          </div>
        </t>
        <t
          key="section_Legal"
          lg={null}
          md={null}
          sm={null}
          tag="div"
          xl={null}
          xs={null}
        >
          <div
            className="col"
          >
            <h4
              style={
                Object {
                  "marginBottom": "15px",
                }
              }
            >
              <b>
                Legal
              </b>
            </h4>
            <div
              className="custom_link"
              key="link_0"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "fontWeight": "400",
                }
              }
            >
              Disclaimer
            </div>
            <div
              className="custom_link"
              key="link_1"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "fontWeight": "400",
                }
              }
            >
              Terms and Conditions
            </div>
            <div
              className="custom_link"
              key="link_2"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "fontWeight": "400",
                }
              }
            >
              Data Licenses
            </div>
            <div
              className="custom_link"
              key="link_3"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "fontWeight": "400",
                }
              }
            >
              Privacy
            </div>
          </div>
        </t>
        <t
          key="section_SAEON ODP"
          lg={null}
          md={null}
          sm={null}
          tag="div"
          xl={null}
          xs={null}
        >
          <div
            className="col"
          >
            <h4
              style={
                Object {
                  "marginBottom": "15px",
                }
              }
            >
              <b>
                SAEON ODP
              </b>
            </h4>
            <div
              className="custom_link"
              key="link_0"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "fontWeight": "400",
                }
              }
            >
              Open Data Platform
            </div>
            <div
              className="custom_link"
              key="link_1"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "fontWeight": "400",
                }
              }
            >
              Contribute
            </div>
            <div
              className="custom_link"
              key="link_2"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "fontWeight": "400",
                }
              }
            >
              Contact Us
            </div>
          </div>
        </t>
        <t
          key="section_Funding"
          lg={null}
          md={null}
          sm={null}
          tag="div"
          xl={null}
          xs={null}
        >
          <div
            className="col"
          >
            <h4
              style={
                Object {
                  "marginBottom": "15px",
                }
              }
            >
              <b>
                Funding
              </b>
            </h4>
            <img
              key="link_0"
              src={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "cursor": "default",
                  "width": "100%",
                }
              }
            />
          </div>
        </t>
      </div>
    </t>
    <br />
  </div>
  <div
    style={
      Object {
        "backgroundColor": "#8EAC85",
        "color": "white",
        "padding": "10px 45px 10px 45px",
      }
    }
  >
    <t
      tag="div"
    >
      <div
        className="row"
      >
        <t
          lg={null}
          md="12"
          sm={null}
          tag="div"
          xl={null}
          xs={null}
        >
          <div
            className="col-md-12"
          >
            <b>
              Created by
              <span>
                 
                <a
                  href="http://www.saeon.ac.za"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                  target="saeon"
                >
                  <strong>
                    SAEON
                  </strong>
                </a>
                 
              </span>
              using
              <span>
                 
                <a
                  href="https://reactjs.org/"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                  target="react"
                >
                  <strong>
                    React
                  </strong>
                </a>
                , 
              </span>
              <span>
                 
                <a
                  href="https://mdbootstrap.com/react"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                  target="mdb"
                >
                  <strong>
                     MDBootstrap
                  </strong>
                </a>
                , 
              </span>
              <span>
                 
                <a
                  href="https://ant.design/"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                  target="antdesign"
                >
                  <strong>
                     Ant Design 
                  </strong>
                </a>
                 
              </span>
              and SAEON Open Data Platform APIs.
            </b>
            <br />
            Copyright Â© 
            2019
          </div>
        </t>
      </div>
    </t>
  </div>
  <t
    autoFocus={true}
    backdrop={true}
    backdropTransitionTimeout={150}
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransitionTimeout={300}
    role="dialog"
    size="fluid"
    style={
      Object {
        "width": "95%",
      }
    }
    toggle={[Function]}
    zIndex={1050}
  />
</Footer>
`;
